iree/compiler/Conversion/LinalgToLLVM/Passes.cpp

```diff
@@ -106,6 +107,9 @@ void buildLLVMTransformPassPipeline(OpPassManager &passManager) {
       Shape::createMaterializeShapeCalculationsPass());
   passManager.addNestedPass<FuncOp>(Shape::createHoistShapeCalculationsPass());
 
+  // TOSA -> Linalg on tensors.
+  tosa::addTosaToLinalgOnTensorsPasses(passManager);
```

Comment:
we should be doing this one step sooner - linalg->llvm/linalg->spirv shouldn't know about tosa, but instead only accept linalg

Reply:
I moved it to be next to early HLOToHLO processing.